docs/data/migration/migration-pickers-v5/migration-pickers-v5.md

```diff
@@ -304,3 +304,93 @@ Component name changes are also reflected in `themeAugmentation`:
    },
  });
 ```
+
+### Replace `toolbar` props by slot
+
+- The `ToolbarComponent` has been replaced by a `Toolbar` component slot. You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props):
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+  -  ToolbarComponent: MyToolbar,
+  +  components={{ Toolbar: MyToolbar }}
+   />
+  ```
+
+- The `toolbarPlaceholder` and `toolbarFormat` props have been moved to the `toolbar` component props slot:
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+  -  toolbarPlaceholder="__"
+  -  toolbarFormat="DD / MM / YYYY"
+  +  componentsProps={{
+  +    toolbar: {
+  +      toolbarPlaceholder: "__",
+  +      toolbarFormat: "DD / MM / YYYY",
+  +    }
+  +  }}
+   />
+  ```
+
+- The `toolbarTitle` prop has been moved to the localization object:
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+  -  toolbarTitle="Title"
+  +  localeText={{ toolbarTitle: "Title" }}
+   />
+  ```
+
+- The toolbar related translation keys have been renamed to better fit their usage:
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+    localeText={{
+  -    datePickerDefaultToolbarTitle: 'Date Picker',
+  +    datePickerToolbarTitle: 'Date Picker',
+
+  -    timePickerDefaultToolbarTitle: 'Time Picker',
+  +    timePickerToolbarTitle: 'Time Picker',
+
+  -    dateTimePickerDefaultToolbarTitle: 'Date Time Picker',
+  +    dateTimePickerToolbarTitle: 'Date Time Picker',
+
+  -    dateRangePickerDefaultToolbarTitle: 'Date Range Picker',
+  +    dateRangePickerToolbarTitle: 'Date Range Picker',
+    }}
+   />
+  ```
+
+- The `onChange` / `openView` props on the toolbar have been renamed to `onViewChange` / `view`
+
+### Replace `tabs` props
+
+- The `hideTabs`, `dateRangeIcon`, and `timeIcon` props have been moved to `tabs` component props:
+
+  ```diff
+   // Same on all other Date Time picker variations
+   <DateTimePicker
+  -  hideTabs={false}
+  -  dateRangeIcon={<LightModeIcon />}
+  -  timeIcon={<AcUnitIcon />}
+  +  componentsProps={{
+  +    tabs: {
+  +      hidden: false,
+  +      dateRangeIcon: <LightModeIcon />,
+  +      timeIcon: <AcUnitIcon />,
+  +    }
+  +  }}
+   />
+  ```
+
+- The `onChange` prop on `DateTimePickerTabs` component has been renamed to `onViewChange` to better fit its usage:
```

Comment:
The actual breaking change would be if user try to set a custom tab component
I would rephrase it the opposite way

Something like

````suggestion
- The `onChange` prop of the `tabs` component slot has been renamed `onViewChange`

```diff
const CustomTabComponent = props => (
  <div>
-  <button onClick={() => props.onChange('day')}>Show day view</button>
+  <button onClick={() => props.onViewChange('day')}>Show day view</button>
  </div>
)

<DateTimePicker
  components={{
    tabs: CustomTabComponent
  }}
/>
```
````

Reply:
Or they have used our publicly exported `DateTimePickerTabs` to build something custom. 🤔 
I'd say that such changes we could include both the current diff and my suggestion—possibly leaving the text the same..? 🤔

Reply:
You are right, we could do both

Reply:
Updated 😉